### PR TITLE
feat: add new afterFinalizePackageTargets hook

### PR DESCRIPTION
**Pull request by MarshallOfSound**
Ref: https://github.com/electron/forge/pull/3006

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -204,6 +210,11 @@ declare namespace electronPackager {
     afterCopyExtraResources?: HookFunction[];
     /** Functions to be called after the prebuilt Electron binary has been extracted to a temporary directory. */
     afterExtract?: HookFunction[];
+    /**
+     * Functions to be called after the final matrix of platform/arch combination is determined.  Use this to
+     * learn what archs/platforms packager is targetting when you pass "all" as a value.
```
Comment (georgexu99) [inline src/index.d.ts:215]: > learn what archs/platforms packager is targetting when you pass "all" as a value.

passing "all" as a value will return the architectures/platforms that packager is currently targetting.

**Comment by malept** [comment]
🚲🏚️ hook name:

- success
- complete
- allDone

Something along those lines.

**Comment by MarshallOfSound** [comment]
@malept this hook is called before any packaging is actually done

File: src/index.d.ts [hunk 2]
```diff
@@ -204,6 +210,11 @@ declare namespace electronPackager {
...
+     */
+    afterFinalizeTargetMatrix?: FinalizeTargetMatrixHookFunction[];
```
Comment (erickzhao) [inline src/index.d.ts:217]:
```suggestion
    afterFinalizePackageTargets?: FinalizeTargetMatrixHookFunction[];
```

The one thing I didn't like about the naming here was that we don't really use the `matrix` terminology elsewhere.

**Comment by malept** [comment]
> this hook is called before any packaging is actually done
 Hrm, I misread the Forge issue. `afterTargetCalculation` or something similar?

File: src/index.d.ts [hunk 3]
```diff
@@ -121,6 +121,12 @@ declare namespace electronPackager {
     callback: (err?: Error | null) => void
   ) => void;
 
+  type TargetDefinition = {
+    arch: TargetArch;
+    platform: TargetPlatform;
+  }
+  type FinalizeTargetMatrixHookFunction = (matrix: TargetDefinition[]) => void;
```
Comment (macdja38) [inline src/index.d.ts:128]: I think this is missing the callback parameter added by hooks.promisifyHooks

**Review by erickzhao — APPROVED** [review]

**Review by VerteDinde — APPROVED** [review]

**Pull request merged**